src/main/java/ru/yandex/practicum/filmorate/storage/dao/film/impl/FilmDaoImpl.java

```diff
@@ -85,12 +82,21 @@ public void deleteFilmById(Integer filmId) {
     }
 
```

Comment:
Предлагаю **ПОКА** вернуть базовую имплементацию метода `getTopFilms(Integer count)` от интерфейса `FilmStorage` и использовать ее в имплементации метода `getTopFilms(Integer count, Integer genreId, Integer year)` от интерфейса `FilmDao` для случая, когда  `genreId == null & year == null`

```suggestion

    @Override
    public List<Film> getTopFilms(Integer count) {
        String sqlQuery = "SELECT f.*, m.name AS mpa_name FROM films AS f " +
                "LEFT JOIN mpa AS m ON f.mpa_id = m.mpa_id " +
                "LEFT JOIN film_likes AS lk ON f.id = lk.film_id " +
                "LEFT JOIN film_genres AS fg ON f.id = fg.film_id " +
                "GROUP BY f.id ORDER BY COUNT(lk.user_id) DESC LIMIT ?";
        return jdbcTemplate.query(sqlQuery, this::mapRowToFilm, count);
    }
```

Reply:
этот случай обрабатывается в методе getTopFilms, если params пуст